Assets/Plugins/iOS/Shopify/ApplePayEventResponse.swift

```diff
@@ -47,43 +47,37 @@ struct ApplePayEventResponse {
     // ----------------------------------
     //  MARK: - Init -
     //
-    init?(serialized: String) {
-
-        guard
-            let data = serialized.data(using: .utf8),
-            let jsonObject = (try? JSONSerialization.jsonObject(with: data)) as? JSON else {
-                return nil
-        }
-
+    static func deserialized(_ serialized: JSON) -> ApplePayEventResponse? {
```

Comment:
That convention is designed to disambiguate methods that **can** potentially be mutating. Such as method that have behaviour on the received. For example `array.append(value)`. This is a mapping function. It takes `X` returns `Y`. This simply cannot be mutating by nature. Think of it as a `map` or `flatMap`.